### DISCUSSION TITLE
Output Files and Placement for find-orientations

### DISCUSSION THREAD

**Discussion by donald-e-boyce** [HEXRD/hexrd#718] [General]
I am proposing some changes to placement of output files in `find-orientations`
because the current placement seems pretty awkward.

`find-orientations` produces several output files, and only one (grains.out)
goes into the analysis directory. The rest go into the current working
directory with long names based on the name of the analysis directory and
possibly the active material.  Then if you run `find-orientations` several
times on different data, you get a big mess in the current (working) directory.
By contrast, `fit-grains` writes everything to the analysis directory.

I propose changing the names and locations of the output files for
`find-orientations` so that everything goes into the analysis directory
with a simpler name.  Here is an example of input:

```
analysis_name: cstudy
working_dir: .

material:
  definitions: materials.h5
  active: ruby

```
If you run `find-orientations` with this input, it wiil create the following
files:

 * cstudy_ruby_eta-ome_maps.npz
 * accepted_orientations_cstudy_ruby.dat
 * find-orientations_cstudy_ruby.log
 * scored_orientations_cstudy_ruby.npz
 * cstudy/grains.out

So it creates the output directory `cstudy`, but it only puts one file there
and creates four other files in the current directory with
`cstudy` as part of their names.

I would like to change it so that the output files would all be in the
`cstudy` directory with names:

 * eta-ome_maps_ruby.npz
 * accepted_orientations_ruby.dat
 * find-orientations_ruby.log
 * scored_orientations_ruby.npz
 * grains.out

I would also like to make two more small changes:

 1. Also write `grains.npz`.
 2. Allow the analysis_name input to be a path (containing "/" characters).

Overall these changes would be pretty easy to implement, and I don't them
as being disruptive to people's workflows.

**Comment by psavery** [comment]
I completely agree with all of. Thanks for the idea, Don!
